Tools/speech_recognizer/GSSpeechRecognitionServer.m

```diff
@@ -0,0 +1,69 @@
+#import "GSSpeechRecognitionServer.h"
+#import "GSSpeechRecognitionEngine.h"
+#import "GSSpeechRecognizer.h"
+#import <Foundation/Foundation.h>
+
+static GSSpeechRecognitionServer *sharedInstance;
+
+@implementation GSSpeechRecognitionServer
+
++ (void)initialize
+{
+  sharedInstance = [self new];
+}
+
++ (void)start
+{
+  NSConnection *connection = [NSConnection defaultConnection];
+  [connection setRootObject: sharedInstance];
+  if (NO == [connection registerName: @"GSSpeechRecognitionServer"])
+    {
+      return;
+    }
+  [[NSRunLoop currentRunLoop] run];
+}
+
++ (id)sharedServer
+{
+  return sharedInstance;
+}
+
+- (id)init
+{
+  if (nil == (self = [super init])) { return nil; }
```

Comment:
Normally we do not use this one line variant.

Reply:
Corrected

Reply:
I don't see the difference.

Reply:
Apologies, I changed it in the speech synthesizer.  Should be corrected now.